Diff/Diff.sublime-syntax

```diff
@@ -1,75 +1,402 @@
 %YAML 1.2
 ---
-# http://www.sublimetext.com/docs/3/syntax.html
+# https://www.sublimetext.com/docs/syntax.html
+# https://www.gnu.org/software/diffutils/manual/diffutils.html
 name: Diff
+scope: source.diff
+version: 2
+
 file_extensions:
   - diff
   - patch
+
 first_line_match: |-
-  (?x)^
-      (===\ modified\ file
-      |==== \s* // .+ \s - \s .+ \s+ ====
-      |Index:[ ]
-      |---\ [^%]
-      |\*\*\*.*\d{4}\s*$
-      |\d+(,\d+)* (a|d|c) \d+(,\d+)* $
-      |diff\ --git[ ]
-      )
+  ^(?x:
+    ===\ modified\ file
+  | ==== \s* // .+ \s - \s .+ \s+ ====
+  | Index:[ ]
+  | Prereq:[ ]
+  | ---\ [^%]
+  | \*\*\*.*\d{4}\s*$
+  | \d+(,\d+)* [adc] \d+(,\d+)* $
+  )
 
-scope: source.diff
 contexts:
   main:
-    - match: '^((\*{15})|(={67})|(-{3}))$\n?'
+    - match: ^(?={{git_first_line}})
+      push: Packages/Git Formats/Git Diff.sublime-syntax#email-first-line
```

Comment:
Note this means to literly include all contexts of _Git Diff_ into this syntax as well causing _Diff_ to be a merge of both, while _Gitt Diff_ being extended from this one.

> By commenting out this line _Diff.rcache_ size is reduced from 38kB to 7kB for instance.

One could ask, why 2 syntaxes than, because 99% of git related diff/patch files use `.patch` file extension and hence will be opened with this _Diff_ syntax, requiring this pattern to be highlighted as git diff.

It's an interesting mix and hack of first_line_match, at least.

Reply:
Haha. In my commit message from the one that adds this, you can see that I wasn't even confident it would work at all. :stuck_out_tongue_winking_eye:

Reply:
Maybe the duplication can be avoided by making an `embed` with an unmatchable escape sequence?

Reply:
At least git diff-only contexts wouldn't be part of normal diff syntax anymore.

If it is a "first line match", maybe the following would make sense to avoid starting "Git Diff" later on in a document?

Chosen `diffs` as it is the main diff context in _Git Diff_.

```yml
contexts:
  main:
    - match: ^(?={{git_first_line}})
      embed: Packages/Git Formats/Git Diff.sublime-syntax#email-first-line
      escape: (?!)
    - match: ^
      set: diffs

  diffs:
    - include: leading-text
    - include: line-ranges
    - include: headers
    - include: deltas
    - include: diff3-normal
```

Reply:
I don't really understand how, but I think [6826deb](https://github.com/sublimehq/Packages/pull/4111/commits/6826deb45c4e7d29a24264dedd8d6102879ed88b) is messing up other syntaxes that embed Diff.

Reply:
Maybe caused by `set: diffs`. `push: diffs` should solve it.